### PR TITLE
Support private hosted zone

### PR DESCRIPTION
**Pull request by jbcom**
## what

* brings #52 up to date with latest master
* allow creation of private hosted zones for internal network @kevcube 
* Misc: no longer support AWS Provider `2.x` @korenyoni

## why

* #52  has merge conflicts
* I want to put my database hostnames inside private DNS @kevcube 
* Misc: We are now one major version ahead of AWS Provider `2.x`. @kevcube 
 
## references

* [terraform route53 hosted zone docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone#vpc_id)
* Closes #52

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,19 +1,27 @@
 locals {
+<<<<<<< HEAD
   enabled                    = module.this.enabled ? 1 : 0
   parent_zone_record_enabled = var.parent_zone_record_enabled && module.this.enabled ? 1 : 0
   zone_name                  = local.parent_zone_record_enabled == 1 ? var.zone_name : replace(var.zone_name, ".$${parent_zone_name}", "")
+=======
+  enabled                    = module.this.enabled
+  parent_zone_record_enabled = local.enabled && var.parent_zone_record_enabled
+  public_zone                = try(length(var.private_hosted_zone_vpc_attachments), 0) == 0
+>>>>>>> kevcube/support_private_hosted_zone
```
Comment (nitrocode) [inline main.tf:10]: Merge issue

File: main.tf [hunk 2]
```diff
@@ -24,14 +32,27 @@ resource "aws_route53_zone" "default" {
...
+  "$${region}", join("", data.aws_region.default.*.name))
+
+  dynamic "vpc" {
+    for_each = var.private_hosted_zone_vpc_attachments
+    content {
+      vpc_id     = vpc.value.vpc_id
+      vpc_region = vpc.value.vpc_region
+    }
+  }
+>>>>>>> kevcube/support_private_hosted_zone
```
Comment (nitrocode) [inline main.tf:49]: Merge issue

File: main.tf (hunk 1 above)
Comment (jbcom) [reply main.tf]: My bad, happened when I was cleaning up the initial merge pull. Resolved.

File: main.tf (hunk 2 above)
Comment (jbcom) [reply main.tf]: My bad, happened when I was cleaning up the initial merge pull. Resolved.

File: main.tf [hunk 3]
```diff
@@ -2,18 +2,21 @@ locals {
...
+data "aws_region" "default" {
+  count = local.enabled ? 1 : 0
+}
 
 data "aws_route53_zone" "parent_zone" {
-  count   = local.parent_zone_record_enabled
+  count   = local.parent_zone_record_enabled ? 1 : 0
   zone_id = var.parent_zone_id
   name    = var.parent_zone_name
 }
```
Comment (nitrocode) [inline main.tf:16]: Can we set private_zone = true here ?

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/route53_zone#private_zone

File: outputs.tf [hunk 4]
```diff
@@ -27,3 +27,8 @@ output "fqdn" {
   value       = join("", aws_route53_zone.default.*.name)
   description = "Fully-qualified domain name"
 }
+
+output "type" {
```
Comment (nitrocode) [inline outputs.tf:31]: Can we make this output a boolean instead?
```suggestion
output "private_zone" {
```

**Pull request closed without merging**